lib/internal/http2.js

```diff
@@ -48,7 +46,139 @@ Object.defineProperty(exports, 'constants', {
 // emit events as callbacks. To allow this to happen,
 // it has to inherit from EventEmitter.
 util.inherits(http2.Http2Session, EventEmitter);
+util.inherits(http2.Http2Stream, Duplex);
 
+function maybeDestroyStream(stream) {
+  if ((!stream.readable &&
+       !stream.writable &&
+       !stream._writableState.length)) {
+    stream.destroy();
+  }
```

Comment:
that's perfect. Can you put the above description as a comment on top of the function? it's definitely not clear :D. It seems the usual pattern "destroy if there is a destroy function", which I have written thousands of time.